pkg/jobrunaggregator/jobrunaggregatoranalyzer/pass_fail.go

```diff
@@ -0,0 +1,436 @@
+		1, // 100
+	}
+	requiredPassesByPassPercentageFor_00_Attempts = []int{
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 00-09
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 10-19
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 20-29
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 30-39
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 40-49
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 50-59
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 60-69
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 70-79
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 80-89
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, // 90-99
+		0, // 100
+	}
+
+	requiredPassesByPassPercentageByNumberOfAttempts = [][]int{
+		requiredPassesByPassPercentageFor_00_Attempts,
+		requiredPassesByPassPercentageFor_01_Attempts,
+		requiredPassesByPassPercentageFor_02_Attempts,
+		requiredPassesByPassPercentageFor_03_Attempts,
+		requiredPassesByPassPercentageFor_04_Attempts,
+		requiredPassesByPassPercentageFor_05_Attempts,
+		requiredPassesByPassPercentageFor_06_Attempts,
+		requiredPassesByPassPercentageFor_07_Attempts,
+		requiredPassesByPassPercentageFor_08_Attempts,
+		requiredPassesByPassPercentageFor_09_Attempts,
+		requiredPassesByPassPercentageFor_10_Attempts,
+	}
+)
+
+func didTestRun(testCaseDetails *TestCaseDetails) bool {
+	if len(testCaseDetails.Passes) == 0 && len(testCaseDetails.Failures) == 0 {
+		return false
+	}
+	return true
+}
+
+func getWorkingPercentage(testCaseDetails *TestCaseDetails) float32 {
+	// if the same job run has a pass and a fail, then it's a flake.  For now, we will consider those as passes by subtracting
+	// them (for now), from the failure count.
+	failureCount := len(testCaseDetails.Failures)
+	for _, failure := range testCaseDetails.Failures {
+		for _, pass := range testCaseDetails.Passes {
+			if pass.JobRunID == failure.JobRunID {
+				failureCount--
+				break
+			}
+		}
+	}
+
+	return float32(len(testCaseDetails.Passes)) / float32(len(testCaseDetails.Passes)+failureCount) * 100.0
+}
+
+func getAttempts(testCaseDetails *TestCaseDetails) int {
+	// if the same job run has a pass and a fail, or multiple passes, it only counts as a single attempt.
+	jobRunsThatAttempted := sets.NewString()
+	for _, failure := range testCaseDetails.Failures {
+		jobRunsThatAttempted.Insert(failure.JobRunID)
+	}
+	for _, pass := range testCaseDetails.Passes {
+		jobRunsThatAttempted.Insert(pass.JobRunID)
+	}
+
+	return len(jobRunsThatAttempted)
+}
+
+func getNumberOfPasses(testCaseDetails *TestCaseDetails) int {
+	// if the same job run has multiple passes, it only counts as a single pass.
+	jobRunsThatPassed := sets.NewString()
+	for _, pass := range testCaseDetails.Passes {
+		jobRunsThatPassed.Insert(pass.JobRunID)
+	}
+
+	return len(jobRunsThatPassed)
+}
+
+func getNumberOfFailures(testCaseDetails *TestCaseDetails) int {
+	// if the same job run has multiple failures, it only counts as a single failure.
+	jobRunsThatFailed := sets.NewString()
+	for _, failure := range testCaseDetails.Failures {
+		jobRunsThatFailed.Insert(failure.JobRunID)
+	}
+
+	return len(jobRunsThatFailed)
```

Comment:
I'd like to get more into this module via unit tests after hitting a few for the uploader. For now I think this looks fine.